_source/_posts/2022-06-19-kubernetes-jhipster-aws.md

```diff
@@ -0,0 +1,778 @@
+
+Gradle and Jib make this super easy. In fact, you have the commands you need **in your console output**. You can copy and paste them from there and run them in the correct directories. You cannot copy and paste the commands from my terminal listing above without replacing the docker repository name.
+
+Make sure you are logged into Docker.
+
+```bash
+docker login
+```
+
+The commands to build the Docker images and push them to Docker Hub are the following. You must replace `{your-docker-repository-name}` with your actual docker repository name and run these commands in the correct project subdirectories.
+
+```bash
+# in ./gateway
+./gradlew bootJar -Pprod jib -Djib.to.image={your-docker-repository-name}/gateway
+# in ./blog
+./gradlew bootJar -Pprod jib -Djib.to.image={your-docker-repository-name}/blog
+# in ./store
+./gradlew bootJar -Pprod jib -Djib.to.image={your-docker-repository-name}/store
+```
+
+You're only building the blog, store, and gateway, not the registry. This is because the registry using a stock image and you don't have to build it.
+
+## Configure OAuth Security with Okta
+
+Before you start, you’ll need a free Okta developer account. Install the [Okta CLI](https://cli.okta.com). Run `okta register` to sign up for a new account. If you already have an account, run `okta login`.
+
+Then, from the root project directory, open a Bash shell and run `okta apps create jhipster`. 
+
+The default app name is fine. Or change it. Also accept the default redirect URIs.
+
+My output looked like this:
+
+```bash
+Application name [kubernetes]: 
+Redirect URI
+Common defaults:
+ Spring Security - http://localhost:8080/login/oauth2/code/okta
+ JHipster - http://localhost:8080/login/oauth2/code/oidc
+ Quarkus OIDC - http://localhost:8080/callback
+Enter your Redirect URI(s) [http://localhost:8080/login/oauth2/code/oidc, http://localhost:8761/login/oauth2/code/oidc]: 
+Enter your Post Logout Redirect URI(s) [http://localhost:8080/, http://localhost:8761/]: 
+Configuring a new OIDC Application, almost done:
+Created OIDC application, client-id: 0oa7pu3434DasvWSo4x7
+Creating Authorization Server claim 'groups':
+Adding user 'andrew.hughes@mail.com' to groups: [ROLE_USER, ROLE_ADMIN]
+Existing group 'ROLE_USER' found
+Existing group 'ROLE_ADMIN' found
+|
+Okta application configuration has been written to: /home/andrewcarterhughes/Development/okta/2022/aws-kubernetes/QA1/kubernetes/.okta.env
+```
+
+This command creates an OIDC application on the Okta servers for you. It also creates a a `groups` claim on your authorization server and adds to groups: `ROLE_USER` and `ROLE_ADMIN`. The group names have `ROLE_` prefixed to them because of the way Spring Boot processes group names. You need these default groups for authentication to work.
```

Comment:
```suggestion
This command creates an OIDC application on the Okta servers for you. It also creates a `groups` claim on your authorization server and adds to groups: `ROLE_USER` and `ROLE_ADMIN`. The group names have `ROLE_` prefixed to them because of the way Spring Boot processes group names. You need these default groups for authentication to work.
```